shim_maker.py

```diff
@@ -1,108 +1,162 @@
-            os.makedirs(f"{shim_path}/{fn}", exist_ok=True)
-            shim_folder(f"{path}/{fn}", f"{pypath}.{fn}", f"{shim_path}/{fn}")
-
-
-def shim(path: str, shim_path: str) -> None:
-    os.makedirs("discord", exist_ok=True)
-    shim_folder(path, path.rstrip("/").split("/")[-1], shim_path)
-
-
-if len(sys.argv) != 2:
-    print(f"Usage: {sys.argv[0]} <module path>", file=sys.stderr)
-    exit(1)
+import isort
 
-shim(sys.argv[1], "discord")
+ISORT_CONFIG = pathlib.Path("pyproject.toml")
+
+
+def find_packages(dir: str, package: str = None) -> List[str]:
+    """
+    Finds all of the submodules in the designated modules and returns a list of them.
+
+    Parameters
+    ----------
+    dir : str
+        The directory to search for packages.
+    package : str, optional
+        The package to search for submodules, defaults to dir.
+    """
+    submodules = set()
+    package = package or dir
+    for path in glob.iglob(f"{dir}/**/*.py", recursive=True):
+        path = path[len(dir) : -3].replace(os.sep, ".")
+        if importlib.util.find_spec(path, package) is not None:
+            submodules.add(path)
+
+    return sorted([submod[:-9] for submod in submodules if submod.endswith("__init__")])
+
+
+def _type_shim_code(code: ModuleType, is_init=False) -> str:
+    """Provided code, returns a type-stub version of it."""
+    shim = ""
+    if code.__doc__:
+        shim += f'"""{code.__doc__}"""\n'
+
+    imports = set()
+
+    # add all non-private attributes
+    def _filter_members(member):
+        if is_init:
+            return inspect.ismodule(member) and member.__name__.startswith(code.__name__)
+        else:
+            return hasattr(member, "__module__") and member.__module__ == code.__name__
+
+    for name, member in inspect.getmembers(code, predicate=_filter_members):
+        if name.startswith("_"):
+            continue
+        if is_init:
+            imports.add(member.__name__[len(code.__name__) + 1 :])
+        else:
+            imports.add(name)
+
+    if hasattr(code, "__all__") and not is_init:
+        imports.update(code.__all__)
+
+        if len(code.__all__) > 1:
+            shim += "__all__ = (\n"
+            shim += textwrap.indent(
+                "\n".join([f'"{member}",' for member in sorted(code.__all__)]), "    "
+            )
+            shim += "\n)\n\n"
+        elif len(code.__all__) == 1:
+            shim += "__all__ = (" + f'"{code.__all__[0]}",' + ")\n\n"
+        else:
+            shim += "__all__ = ()\n\n"
+
+    if is_init:
+        shim += f"from {code.__name__} import *\n"
+        for mod in imports:
+            shim += f"from .{mod} import *\n"
+    else:
+        shim += f"from {code.__name__} import {', '.join(sorted(imports))}\n"
+
+    shim = isort.api.sort_code_string(shim, file_path=ISORT_CONFIG)
+    return shim.strip() + "\n"
+
+
+def _shim_module_type(
+    base_module: ModuleType, shim_module: str, *, original_shim: pathlib.Path = None
+):
```

Comment:
```suggestion
def _shim_module_type(
    base_module: ModuleType, shim_module: str
) -> None:
```
all the other `original_shim` code below is unused as well